cube/src/main/java/zone/cogni/asquare/cube/index/ElasticsearchIndexingService.java

```diff
@@ -0,0 +1,604 @@
+    }
+
+    if (!collectionNames.isEmpty())
+      log.info("{}   valid collections:   {}", indent, String.join(", ", collectionNames));
+    if (!getInvalidCollectionNames(indexMetadata).isEmpty())
+      log.info("{}   invalid collections: {}", indent, String.join(", ", getInvalidCollectionNames(indexMetadata)));
+
+    // deeper check into collections
+    indexMetadata.getCollections().forEach(this::validateCollection);
+  }
+
+  private void validateCollection(CollectionMetadata collectionMetadata) {
+    // valid or not
+    if (collectionMetadata.isValid())
+      log.info("    '{}' collection is valid", collectionMetadata.getName());
+    else
+      log.error("    '{}' collection NOT is valid", collectionMetadata.getName());
+
+    // select
+    if (collectionMetadata.getSelectQueryResources().isEmpty())
+      log.error("{}     select-* queries are missing", indent);
+
+    // construct
+    if (collectionMetadata.getConstructQueryResources().isEmpty())
+      log.error("{}     construct-* queries are missing", indent);
+
+    // facet
+    if (collectionMetadata.getFacetQueryResources().isEmpty())
+      log.warn("{}     facets/* queries are missing", indent);
+  }
+
+  /**
+   * @return <code>true</code> if indexing is active
+   */
+  public boolean isIndexRunning() {
+    return indexMonitoredPool.isActive();
+  }
+
+  /**
+   * @return list of indexes managed by service for current <code>elasticStore</code>
+   */
+  @Nonnull
+  public List<String> getIndexNames() {
+    return indexesMetadata.stream()
+                          .filter(IndexMetadata::isValid)
+                          .map(IndexMetadata::getName)
+                          .collect(Collectors.toList());
+  }
+
+  private List<String> getInvalidIndexNames() {
+    return indexesMetadata.stream()
+                          .filter(indexMetadata -> !indexMetadata.isValid())
+                          .map(IndexMetadata::getName)
+                          .collect(Collectors.toList());
+  }
+
+
+  /**
+   * Fill all indexes.
+   * Can do a clear index in <code>clear</code> parameter is set to <code>true</code>.
+   *
+   * @param clear clears index before filling it again
+   */
+  public void indexAll(boolean clear) {
+    indexByName(getIndexNames(), clear);
+  }
+
+  /**
+   * Fills listed indexes.
+   * Can do a clear index in <code>clear</code> parameter is set to <code>true</code>.
+   *
+   * @param indexes list of indexes to be filled
+   * @param clear   clears index before filling it again
+   */
+  public void indexByName(@Nonnull List<String> indexes, boolean clear) {
+    indexes.forEach(name -> indexByName(name, clear));
+  }
+
+  /**
+   * Fills a single index.
+   * Can do a clear index in <code>clear</code> parameter is set to <code>true</code>.
+   *
+   * @param index to be filled
+   * @param clear clears index before filling it again
+   */
+  private void indexByName(@Nonnull String index, boolean clear) {
```

Comment:
should be public i guess

Reply:
not sure. for APIs it's not needed. they can easily use the list solution. but if we think it could be useful I can do it.

Reply:
for maintenance this can be useful (ofc if there is an api calling that), so i propose to make it public

Reply:
yes. I changed it. strange that it does not show up here.